src/InstrumentationEngine/ProfilerManagerWrapper.cpp

```diff
@@ -0,0 +1,197 @@
+ */
+HRESULT CProfilerManagerWrapper::GetLoggingInstance(_Out_ IProfilerManagerLogging** ppLogging)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IProfilerManagerLogging> pProfilerManagerLogging = (IProfilerManagerLogging*)this;
+    *ppLogging = pProfilerManagerLogging.Detach();
+    return S_OK;
+}
+
+HRESULT CProfilerManagerWrapper::SetLoggingHost(_In_opt_ IProfilerManagerLoggingHost* pLoggingHost)
+{
+    return m_pProfilerManager->SetLoggingHost(pLoggingHost);
+}
+
+HRESULT CProfilerManagerWrapper::GetAppDomainCollection(_Out_ IAppDomainCollection** ppAppDomainCollection)
+{
+    return m_pProfilerManager->GetAppDomainCollection(ppAppDomainCollection);
+}
+
+
+HRESULT CProfilerManagerWrapper::CreateSignatureBuilder(_Out_ ISignatureBuilder** ppSignatureBuilder)
+{
+    return m_pProfilerManager->CreateSignatureBuilder(ppSignatureBuilder);
+}
+
+HRESULT CProfilerManagerWrapper::GetInstrumentationMethod(_In_ REFGUID cslid, _Out_ IUnknown** ppUnknown)
+{
+    return m_pProfilerManager->GetInstrumentationMethod(cslid, ppUnknown);
+}
+
+HRESULT CProfilerManagerWrapper::RemoveInstrumentationMethod(_In_ IInstrumentationMethod* pInstrumentationMethod)
+{
+    return m_pProfilerManager->RemoveInstrumentationMethod(pInstrumentationMethod);
+}
+
+HRESULT CProfilerManagerWrapper::AddInstrumentationMethod(
+    _In_ BSTR bstrModulePath,
+    _In_ BSTR bstrName,
+    _In_ BSTR bstrDescription,
+    _In_ BSTR bstrModule,
+    _In_ BSTR bstrClassGuid,
+    _In_ DWORD dwPriority,
+    _Out_ IInstrumentationMethod** ppInstrumentationMethod)
+{
+    return m_pProfilerManager->AddInstrumentationMethod(
+        bstrModulePath,
+        bstrName,
+        bstrDescription,
+        bstrModule,
+        bstrClassGuid,
+        dwPriority,
+        ppInstrumentationMethod);
+}
+
+// IProfilerManager2 Methods
+HRESULT CProfilerManagerWrapper::DisableProfiling()
+{
+    return m_pProfilerManager->DisableProfiling();
+}
+
+HRESULT CProfilerManagerWrapper::ApplyMetadata(_In_ IModuleInfo* pMethodInfo)
+{
+    return m_pProfilerManager->ApplyMetadata(pMethodInfo);
+}
+
+// IProfilerManager3 Methods
+HRESULT CProfilerManagerWrapper::GetApiVersion(_Out_ DWORD* pApiVer)
+{
+    return m_pProfilerManager->GetApiVersion(pApiVer);
+}
+
+// IProfilerManager4 Methods
+HRESULT CProfilerManagerWrapper::GetGlobalLoggingInstance(_Out_ IProfilerManagerLogging** ppLogging)
+{
+    return m_pProfilerManager->GetGlobalLoggingInstance(ppLogging);
+}
+
+
+// IProfilerManager5 Methods
+HRESULT CProfilerManagerWrapper::IsInstrumentationMethodRegistered(_In_ REFGUID clsid, _Out_ BOOL* pfRegistered)
+{
+    return m_pProfilerManager->IsInstrumentationMethodRegistered(clsid, pfRegistered);
+}
+
+// IProfilerManagerLogging Methods
+/*
+ * Inject Instrumentation Method ClassId
+ */
+HRESULT CProfilerManagerWrapper::LogMessage(_In_ const WCHAR* wszMessage)
+{
+    HRESULT hr = S_OK;
+    WCHAR wszBuffer[LogEntryMaxSize] = _T("[IM:%s]");
+    IfFailRetErrno(wcscat_s(wszBuffer, wszMessage));
```

Comment:
This is assuming that there are no residual format specifiers or that the caller was intentionally try to log something that looks like a format specified. Do we need to escape those, or do some kind of replacement before/after formatting the string?

Reply:
Good point, This interface does not assume the string has format specifiers, I'll have to escape `%` in the original string.

---
In reply to: [333779838](https://github.com/microsoft/CLRInstrumentationEngine/pull/146#discussion_r333779838) [](ancestors = 333779838)